src/scheduler/scheduler.ts

```diff
@@ -128,11 +128,18 @@ class DetachableLoop {
 
             if (sleepTime > 0) {
                 // Wait for either the timeout or an interruption
+                let timer: NodeJS.Timeout;
+                const timeoutPromise = new Promise<void>((resolve) => {
+                    timer = setTimeout(() => {
+                      resolve();
+                    }, sleepTime);
+                  });
                 await Promise.race([
-                    this.sleepms(sleepTime),
-                    new Promise<void>((_, reject) => this.interruptResolve = reject)
+                    timeoutPromise,
+                    new Promise<void>((resolve) => this.interruptResolve = resolve)
```

Comment:
What does this line do?  Why was the function from (resolve, reject) and is now just from (resolve)?

Reply:
The previous implementation didn't work as expected. `reject` doesn't fulfill `Promise.race`. Using resolve can properly interrupt the loop.

I found we could use `new Promise<void>((_, reject) => this.interruptResolve = reject).catch(() => {})` but it's uglier

Reply:
Found a better solution.